### PR TITLE
tests: Add test for drivers/mm/mm_drv_bank module

### PR DESCRIPTION
**Pull request by peter-mitsis**
Adds a test for the drivers/mm/mm_drv_bank module.

### CODE REVIEW

**Review by dcpleung — DISMISSED** [review]

File: tests/drivers/mm/sys_mm_drv_bank/testcase.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+tests:
+  drivers.mm.sys_mm_drv_bank:
```
Comment (galak) [inline tests/drivers/mm/sys_mm_drv_bank/testcase.yaml:2]: Seems like we should have some filter so we only build / run this on platforms that have drivers/mm.

**Review by galak — CHANGES_REQUESTED** [review]
Need to filter this for platforms that actually have a "driver" that exists.  Otherwise we get build failures on arm-clang.

File: tests/drivers/mm/sys_mm_drv_bank/testcase.yaml (hunk 1 above)
Comment (galak) [reply tests/drivers/mm/sys_mm_drv_bank/testcase.yaml:2]: with arm-clang we get errors of the form:
```
Error: L6218E: Undefined symbol sys_mm_drv_page_phys_get (referred from mm_drv_common.o).
Error: L6218E: Undefined symbol sys_mm_drv_map_page (referred from mm_drv_common.o).
Error: L6218E: Undefined symbol sys_mm_drv_unmap_page (referred from mm_drv_common.o).
Error: L6218E: Undefined symbol sys_mm_drv_page_flag_get (referred from mm_drv_common.o).
Error: L6218E: Undefined symbol sys_mm_drv_update_page_flags (referred from mm_drv_common.o).
```

File: tests/drivers/mm/sys_mm_drv_bank/testcase.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+    tags: drivers mm
+    filter: CONFIG_MM_DRV
```
Comment (galak) [inline tests/drivers/mm/sys_mm_drv_bank/testcase.yaml:4]: `filter: CONFIG_MM_DRV` does not solve the issue.   I think adding something like:

```
filter: dt_compat_enabled("intel,adsp-tlb") or dt_compat_enabled("intel,adsp-mtl-tlb")
```

We need to add this to `tests/drivers/mm/sys_mm_drv_api/testcase.yaml`

Comment (peter-mitsis) [reply tests/drivers/mm/sys_mm_drv_bank/testcase.yaml]: Gotcha. Hopefully this will do it. :)

**Review by galak — DISMISSED** [review]

File: tests/drivers/mm/sys_mm_drv_api/testcase.yaml [hunk 3]
```diff
@@ -1,3 +1,4 @@
 tests:
   drivers.mm.sys_mm_drv_api:
     tags: drivers mm
+    filter: dt_compat_enabled("intel,adsp-mtl-tlb")
```
Comment (galak) [inline tests/drivers/mm/sys_mm_drv_api/testcase.yaml:4]: wny not `or dt_compat_enabled("intel,adsp-tlb")`

File: tests/drivers/mm/sys_mm_drv_bank/testcase.yaml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+tests:
+  drivers.mm.sys_mm_drv_bank:
+    tags: drivers mm
+    filter: dt_compat_enabled("intel,adsp-mtl-tlb")
```
Comment (galak) [inline tests/drivers/mm/sys_mm_drv_bank/testcase.yaml:4]: wny not or dt_compat_enabled("intel,adsp-tlb")

**Review by galak — APPROVED** [review]

**Review by dcpleung — APPROVED** [review]

**Pull request merged**